scriptmodules/supplementary/emulationstation/configscripts/mupen64plus.sh

```diff
@@ -0,0 +1,266 @@
+# 
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+# 
+# See the LICENSE.md file at the top-level directory of this distribution and 
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+function onstart_mupen64plus_joystick() {
+    local device_type=$1
+    local device_name=$2
+
+    # write temp file header
+    echo "; ${device_name}_START " > /tmp/mp64tempconfig.cfg
+    echo "[${device_name}]" >> /tmp/mp64tempconfig.cfg
+    iniConfig " = " "" "/tmp/mp64tempconfig.cfg"
+    iniSet "plugged" "True"
+    iniSet "plugin" "2"
+    iniSet "mouse" "False"
+    iniSet "AnalogDeadzone" "4096,4096"
+    iniSet "AnalogPeak" "32768,32768"
+    iniSet "Mempak switch" ""
+    iniSet "Rumblepak switch" ""
+}
+
+function map_mupen64plus_joystick() {
+    local device_type="$1"
+    local device_name="$2"
+    local input_name="$3"
+    local input_type="$4"
+    local input_id="$5"
+    local input_value="$6"
+
+    local keys
+    local dir
+    case "$input_name" in
+        up)
+            keys=("DPad U")
+            ;;
+        down)
+            keys=("DPad D")
+            ;;
+        left)
+            keys=("DPad L")
+            ;;
+        right)
+            keys=("DPad R")
+            ;;
+        b)
+            keys=("A Button")
+            ;;
+        y)
+            keys=("B Button")
+            ;;
+        a)
+            keys=("C Button D")
+            ;;
+        x)
+            keys=("C Button U")
+            ;;
+        leftbottom)
+            keys=("Z Trig")
+            ;;
+        rightbottom)
+            keys=("R Trig")
+            ;;
+        lefttop)
+            keys=("L Trig")
+            ;;
+        start)
+            keys=("Start")
+            ;;
+        leftanalogleft)
+            keys=("X Axis")
+            dir=("Left")
+            ;;
+        leftanalogright)
+            keys=("X Axis")
+            dir=("Right")
+            ;;
+        leftanalogup)
+            keys=("Y Axis")
+            dir=("Up")
+            ;;
+        leftanalogdown)
+            keys=("Y Axis")
+            dir=("Down")
+            ;;
+        rightanalogleft)
+            keys=("C Button L")
+            ;;
+        rightanalogright)
+            keys=("C Button R")
+            ;;
+        rightanalogup)
+            keys=("C Button U")
+            ;;
+        rightanalogdown)
+            keys=("C Button D")
+            ;;
+        *)
+            return
+            ;;
+    esac
+
+    local key
+    local value
+    #iniConfig " = " "" "/tmp/mp64keys.cfg"
+    for key in "${keys[@]}"; do
+        # read key value. Axis takes two key/axis values.
+        iniGet "$key"
+        if [[ "$input_type" == "axis" ]]; then
+            # key "X/Y Axis" needs different button naming
+            if echo "$key" | grep -q "Axis"; then
```

Comment:
in some of the cases where you need a simple match it might be cleaner to use case / esac with wildcards etc.